app/views/_order_items.html.erb

```diff
@@ -1,28 +1,36 @@
 <% @order_items.each_with_index do |order_item, i| %>
-  <div id="item-<%= order_item.item_id %>" class='show-order-item'>
-    <h2><span id='item-number'>Item <%= i + 1 %>:</span></h2>
-      <%= image_tag(order_item.item.image, class: "thumb", onerror: 'this.src="/no_image_available.jpg"') %>
-      <ul>
-        <li><b>Name: </b><%= link_to "#{order_item.item.name.titleize}", item_path(order_item.item) %></li>
-        <li><b>Description: </b><%= order_item.item.description %></li>
-        <li><b>Quantity: </b><%= order_item.quantity %></li>
-        <li><b>Price: </b><%= number_to_currency(order_item.price) %></li>
-        <li><b>Subtotal: </b><%= number_to_currency(order_item.subtotal) %></li>
-        <li><b>Status: </b>
-          <% if order_item.fulfilled? %>
-              <%= "Fulfilled" %>
-            <% else %>
-              <%= "Not Fulfilled" %>
-          <% end %></li>
-        <li>
-            <% if order_item.fulfillable? && merchant_user? || order_item.fulfillable? && admin_user?%>
-              <% if current_user && current_user.merchant? %>
-                <%= button_to 'Fulfill', dashboard_order_item_fulfill_path(order_item), method: 'patch' %>
-              <% elsif current_user && current_user.admin? %>
+  <div class='order-item-container'>
+    <div id="item-<%= order_item.item_id %>" class='show-order-item'>
+      <h2><span id='item-number'>Item <%= i + 1 %>:</span></h2>
+        <%= image_tag(order_item.item.image, class: "thumb", onerror: 'this.src="/no_image_available.jpg"') %>
+        <ul>
+          <li><b>Name: </b><%= link_to "#{order_item.item.name.titleize}", item_path(order_item.item) %></li>
+          <li><b>Description: </b><%= order_item.item.description %></li>
+          <li><b>Quantity: </b><%= order_item.quantity %></li>
+          <li><b>Price: </b><%= number_to_currency(order_item.price) %></li>
+          <li><b>Subtotal: </b><%= number_to_currency(order_item.subtotal) %></li>
+          <li><b>Status: </b>
+            <% if order_item.fulfilled? %>
+                <%= "Fulfilled" %>
+              <% else %>
+                <%= "Not Fulfilled" %>
+            <% end %></li>
+          <li>
+            <% if current_user && current_user.merchant? %>
+              <% if order_item.fulfillable? %>
+                <%= button_to 'Fulfill', dashboard_order_item_fulfill_path(order_item), method: 'patch'%>
+              <% elsif order_item.quantity > order_item.item.instock_qty %>
+                <div class='out-of-stock'>Out of Stock</div>
+              <% end %>
```

Comment:
This works, but it could probably be drier. My recommendation (from a previous conversation) is setting a variable in the controllers that contains the name of the path helper for the button as a symbol, and using public_send to call it.

One thing I noticed while I was walking through is that the "Out of Stock" appears centered when viewed by an admin, but not when viewed by a merchant. I'm not sure why that is, luckily it's not too big of a problem.

I ran through it on my local production environment (had to remember to run rake assets:precompile haha). Everything looked good. Suggests it will be good on Heroku.

Reply:
I'm not sure why it appears not centered as a merchant. I definitely agree on the code needing to be more DRY. I'll put up a waffle card for it.